src/middlewared/middlewared/utils/io.py

```diff
@@ -8,7 +8,7 @@ def write_if_changed(path, data):
 
     changed = False
 
-    with open(os.open(path, os.O_CREAT | os.O_RDWR), 'wb+') as f:
+    with open(path, 'wb+') as f:
```

Comment:
Every doc says the contrary, and this still does not work
```
root@truenas[~]# echo "long text" >> file
root@truenas[~]# python                  
Python 3.8.5 (default, Dec  3 2020, 23:15:58) 
[Clang 10.0.1 (git@github.com:llvm/llvm-project.git llvmorg-10.0.1-0-gef32c611a on freebsd12
Type "help", "copyright", "credits" or "license" for more information.
>>> open('file').read()
'long text\n'
>>> f = open('file', 'wb+')
>>> f.seek(0)
0
>>> f.read()
b''
```

Reply:
It was truncated because of the `w` option.

Reply:
`>>> f = open('testing.txt', 'wb+')
>>> f.write(b'neat')
4
>>> f.flush()
>>> f.seek(0)
0
>>> f.read()
b'neat'`

Reply:
@yocalebo that wont work on our use case because the file already exists. So the issue was it was getting truncated, not read+write

Reply:
The real problem, actually, is that `os.open` is missing the `mode` option.